src/main/java/org/opensearch/security/ssl/SslSettingsManager.java

```diff
@@ -0,0 +1,395 @@
+            );
+        Optional.ofNullable(configurations.get(SSL_TRANSPORT_PREFIX)).ifPresentOrElse(sslConfiguration -> {
+            contexts.put(SSL_TRANSPORT_PREFIX, new SslContextHandler(sslConfiguration));
+            final var transportClientConfiguration = Optional.ofNullable(configurations.get(SSL_TRANSPORT_CLIENT_PREFIX))
+                .orElse(sslConfiguration);
+            contexts.put(SSL_TRANSPORT_CLIENT_PREFIX, new SslContextHandler(transportClientConfiguration, true));
+        }, () -> LOGGER.warn("SSL Configuration for Transport Layer hasn't been set"));
+        return contexts.build();
+    }
+
+    public synchronized void reloadSslContext(final String sslConfigPrefix) {
+        sslContextHandler(sslConfigPrefix).ifPresentOrElse(sscContextHandler -> {
+            LOGGER.info("Reloading {} SSL context", sslConfigPrefix);
+            try {
+                sscContextHandler.reloadSslContext();
+            } catch (CertificateException e) {
+                throw new OpenSearchException(e);
+            }
+            LOGGER.info("{} SSL context reloaded", sslConfigPrefix);
+        }, () -> LOGGER.error("Missing SSL Context for {}", sslConfigPrefix));
+    }
+
+    private Map<String, SslConfiguration> loadConfigurations(final Environment environment) {
+        final var settings = environment.settings();
+        final var httpSettings = settings.getByPrefix(SSL_HTTP_PREFIX);
+        final var transpotSettings = settings.getByPrefix(SSL_TRANSPORT_PREFIX);
+        if (httpSettings.isEmpty() && transpotSettings.isEmpty()) {
+            throw new OpenSearchException("No SSL configuration found");
+        }
+        jceWarnings();
+        openSslWarnings(settings);
+
+        final var httpEnabled = httpSettings.getAsBoolean(ENABLED, SECURITY_SSL_HTTP_ENABLED_DEFAULT);
+        final var transportEnabled = transpotSettings.getAsBoolean(ENABLED, SECURITY_SSL_TRANSPORT_ENABLED_DEFAULT);
+
+        final var configurationBuilder = ImmutableMap.<String, SslConfiguration>builder();
+        if (httpEnabled && !clientNode(settings)) {
+            validateHttpSettings(httpSettings);
+            final var httpSslParameters = SslParameters.loader(httpSettings).load(true);
+            final var httpTrustAndKeyStore = new SslCertificatesLoader(SSL_HTTP_PREFIX).loadConfiguration(environment);
+            configurationBuilder.put(
+                SSL_HTTP_PREFIX,
+                new SslConfiguration(httpSslParameters, httpTrustAndKeyStore.v1(), httpTrustAndKeyStore.v2())
+            );
+            LOGGER.info("TLS HTTP Provider             : {}", httpSslParameters.provider());
+            LOGGER.info("Enabled TLS protocols for HTTP layer : {}", httpSslParameters.allowedProtocols());
+        }
+        final var transportSslParameters = SslParameters.loader(transpotSettings).load(false);
+        if (!transportEnabled && !sslOnly(settings)) {
```

Comment:
Why the check for `!sslOnly(settings)` here?

Reply:
This is because of the following test: https://github.com/opensearch-project/security/blob/31cfd54a4166af232fb1d584720de29fc0620da0/src/test/java/org/opensearch/security/ssl/SSLTest.java#L576. I'm not entirely sure the test is correct, as it's possible to enable HTTPs without requiring transport TLS.